app/assets/javascripts/discourse/app/controllers/sidebar-section-form.js

```diff
@@ -66,14 +81,15 @@ class SectionLink {
   @tracked value;
   @tracked _destroy;
 
-  constructor({ router, icon, name, value, id }) {
+  constructor({ router, icon, name, value, id, objectId }) {
     this.router = router;
     this.icon = icon || "link";
     this.name = name;
     this.value = value;
     this.id = id;
     this.httpHost = "http://" + window.location.host;
     this.httpsHost = "https://" + window.location.host;
+    this.objectId = objectId;
```

Comment:
generated objectId is required for form link as unique identifier to define what is reorder and what it is dropped. This is especially crucial for new rows